lib/vty.c

```diff
@@ -2207,6 +2235,34 @@ static int vtysh_do_pass_fd(struct vty *vty)
 	return BUFFER_EMPTY;
 }
 
+/* This function is similar to vtysh_flush exceot this is used
+ * to chunkify the vty_outs intermediately when the buffer size
+ * is 128KB or more (since only 131 KB is flushed at once).
+ *
+ * When a BUFFER_EMPTY is received, we clean up the buffer data
+ * When a BUFFER_PENDING is received, we schedule to call the same
+ * function again to process the data in the pending buffer
+ */
+static void vtysh_flush_intermediate(struct vty *vty)
+{
+	int ret;
+
+	ret = buffer_flush_available(vty->obuf, vty->wfd);
+	switch (ret) {
+	case BUFFER_EMPTY:
+		vty->vty_buf_size_written = 0;
+		buffer_reset(vty->lbuf);
+		buffer_reset(vty->obuf);
+		break;
+	case BUFFER_PENDING:
+		vty_event(VTYSH_WRITE_INTERMEDIATE, vty);
+		break;
+	case BUFFER_ERROR:
```

Comment:
why's the error-handling different here than in vtysh_flush()? (if this goes away, no need to answer of course)